### PR TITLE
peer: Add mix message summary strings

### PR DESCRIPTION
**Pull request by jrick**
This includes the message hash and the sender's identity in the short human-readable description of peer messages as they are debug logged.

While here and because mixing introduced a new inventory vector type, add it to the inv summary case statement.

### CODE REVIEW

File: peer/log.go [hunk 1]
```diff
@@ -68,6 +68,8 @@ func invSummary(invList []*wire.InvVect) string {
 			return fmt.Sprintf("tx %s", iv.Hash)
 		case wire.InvTypeFilteredBlock:
 			return fmt.Sprintf("filtered block %s", iv.Hash)
+		case wire.InvTypeMix:
+			return fmt.Sprintf("mix message %s", iv.Hash)
```
Comment (davecgh) [inline peer/log.go:72]: Should also hit the case when there are multiple requests in the same inv below.  e.g.

```diff
diff --git a/peer/log.go b/peer/log.go
index d142f5d9..aa856dec 100644
--- a/peer/log.go
+++ b/peer/log.go
@@ -74,18 +74,21 @@ func invSummary(invList []*wire.InvVect) string {
        }
 
        // More than one inv item.
-       var numTxns, numBlocks uint64
+       var numTxns, numBlocks, numMixes uint64
        for _, iv := range invList {
                switch iv.Type {
                case wire.InvTypeTx:
                        numTxns++
                case wire.InvTypeBlock:
                        numBlocks++
+               case wire.InvTypeMix:
+                       numMixes++
                }
        }
 
-       diff := uint64(invLen) - (numTxns + numBlocks)
-       return fmt.Sprintf("txns %d, blocks %d, other %d", numTxns, numBlocks, diff)
+       diff := uint64(invLen) - (numTxns + numBlocks + numMixes)
+       return fmt.Sprintf("txns %d, blocks %d, mixmsgs %d, other %d", numTxns,
+               numBlocks, numMixes, diff)
 }
```

**Comment by jrick** [comment]
we might want to add to this message as well:

```
return fmt.Sprintf("txns %d, blocks %d, other %d", numTxns, numBlocks, diff)
```

File: peer/log.go (hunk 1 above)
Comment (jrick) [reply peer/log.go:72]: i had the same diff, word for word.

File: peer/log.go [hunk 2]
```diff
@@ -133,6 +148,23 @@ func messageSummary(msg wire.Message) string {
 		return fmt.Sprintf("hash %s, ver %d, %d tx, %s", msg.BlockHash(),
 			header.Version, len(msg.Transactions), header.Timestamp)
 
+	case *wire.MsgMixPairReq:
+		return mixMessageSummary(msg)
+	case *wire.MsgMixKeyExchange:
+		return mixMessageSummary(msg)
```
Comment (davecgh) [inline peer/log.go:154]: These are definitely an improvement over the lack of anything that is there now, but I think we should put a few more details in the `!MsgMixPairReq` cases.

For example, maybe the run number and session id would be nice.

Then, for `MsgMixCiphertexts` perhaps the number of ciphertexts and `MsgMixDCNet` the number of mix vectors, etc.

Comment (jrick) [reply peer/log.go:154]: session/run we can do, the rest will make the "summary" just way too verbose i think. after adding the session we are already at 2 hashes and a serialized pubkey.

Comment (davecgh) [reply peer/log.go:161]: Fair point.  They're pretty long.  Looks better with the session and run, imo.  I already say a `run 1` in my logs with this update.

**Review by davecgh — APPROVED** [review]

**Pull request merged**